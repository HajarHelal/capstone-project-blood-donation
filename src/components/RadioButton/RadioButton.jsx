import React from 'react';
import { PropTypes } from 'prop-types';

const RadioButton = ({ donor, onChange, radiotext, RadioOptions, name }) => (
  <div className="block mb-8" value={donor[name]} name={name} onChange={onChange}>
    <span className="text-sm sm:text-base md:text-lg font-medium inline-block mb-2">
      {radiotext}
    </span>
    {RadioOptions.map((option) => (
      <div key={option} className="mt-2">
        <label htmlFor="sa" className="inline-flex items-center">
          <input
            type="radio"
            className="form-radio text-green-500 h-6 w-6"
            name={name}
            value={option}
            defaultChecked={donor[name] === option}
          />
          <span className="mx-4">{option}</span>
        </label>
      </div>
    ))}
  </div>
);
RadioButton.propTypes = {
  name: PropTypes.string.isRequired,
  radiotext: PropTypes.string.isRequired,
  donor: PropTypes.object.isRequired,
  RadioOptions: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default RadioButton;
