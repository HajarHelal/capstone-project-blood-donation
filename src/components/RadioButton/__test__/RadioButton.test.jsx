import React from 'react';
import renderer from 'react-test-renderer';
import RadioButton from '../RadioButton';

// This is an example of a Snapshot. It will create a new snapshot which all future
// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer
    .create(
      <RadioButton
        donor={{
          name: '',
          email: '',
          phone: '',
          date: '',
          weight: '',
          sickness: '',
          bloodType: '',
          gender: '',
        }}
        onChange={() => console.log('saleem')}
        radiotext="3 . Have you ever had a cancer other than basal cell carcinoma or cervical carcinoma insitu (CIN)?*"
        RadioOptions={['Yes', 'No']}
        name="sickness"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
