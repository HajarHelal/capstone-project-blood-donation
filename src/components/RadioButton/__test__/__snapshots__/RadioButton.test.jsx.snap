// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="block mb-8"
  name="sickness"
  onChange={[Function]}
  value=""
>
  <span
    className="text-sm sm:text-base md:text-lg font-medium inline-block mb-2"
  >
    3 . Have you ever had a cancer other than basal cell carcinoma or cervical carcinoma insitu (CIN)?*
  </span>
  <div
    className="mt-2"
  >
    <label
      className="inline-flex items-center"
      htmlFor="sa"
    >
      <input
        className="form-radio text-green-500 h-6 w-6"
        name="sickness"
        type="radio"
        value="Yes"
      />
      <span
        className="ml-4"
      >
        Yes
      </span>
    </label>
  </div>
  <div
    className="mt-2"
  >
    <label
      className="inline-flex items-center"
      htmlFor="sa"
    >
      <input
        className="form-radio text-green-500 h-6 w-6"
        name="sickness"
        type="radio"
        value="No"
      />
      <span
        className="ml-4"
      >
        No
      </span>
    </label>
  </div>
</div>
`;
